app/scripts/migrations/045.js

```diff
@@ -0,0 +1,23 @@
+const version = 45
+import { cloneDeep } from 'lodash'
+
+/**
+ * Replace ipfsGateway by 'dweb.link'.
+ */
+export default {
+  version,
+  migrate: async function (originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData)
+    versionedData.meta.version = version
+    const state = versionedData.data
+    versionedData.data = transformState(state)
+    return versionedData
+  },
+}
+
+function transformState (state) {
+  if (state.PreferencesController?.ipfsGateway) {
+    state.PreferencesController.ipfsGateway = 'dweb.link'
```

Comment:
IIUC this means user is unable to customize their gateway and is forced to use  `dweb.link`
Can we change this so `ipfsGateway` is set to `dweb.link` only if empty/missing or equal `ipfs.io` or `ipfs.dweb.link` ? And add tests to ensure it works as expected?

Giving users a choice to use custom gateways is very important.